File: datagen/src/main/java/juuxel/adorn/datagen/ConditionType.java
package juuxel.adorn.datagen;

import java.util.Arrays;
import java.util.Map;
import java.util.function.Function;
import java.util.stream.Collectors;

public enum ConditionType {
    NONE("none", null, null, Map.of()),
    FABRIC("fabric", null, null, Map.of("load-condition", "fabric-conditions.json", "model_condition", "fabric-model-conditions.json")),
    FORGE("forge", null, null, Map.of("load-condition", "forge-conditions.json", "model_condition", "forge-model-conditions.json")),
    ;

    private static final Map<String, ConditionType> BY_ID = Arrays.stream(values())
        .collect(Collectors.toMap(type -> type.id, Function.identity()));

    private final String id;
    private final String separateFilePathTemplate;
    private final String separateFileTemplatePath;
    private final Map<String, String> conditionsInFileTemplatePathsByType;

    ConditionType(String id, String separateFilePathTemplate, String separateFileTemplatePath, Map<String, String> conditionsInFileTemplatePathsByType) {
        this.id = id;
        this.separateFilePathTemplate = separateFilePathTemplate;
        this.separateFileTemplatePath = separateFileTemplatePath;
        this.conditionsInFileTemplatePathsByType = conditionsInFileTemplatePathsByType;
    }

    public String getSeparateFilePathTemplate() {
        return separateFilePathTemplate;
    }

    public String getSeparateFileTemplatePath() {
        return separateFileTemplatePath;
    }

    public Map<String, String> getConditionsInFileTemplatePathsByType() {
        return conditionsInFileTemplatePathsByType;
    }

    public static ConditionType parse(String id) {
        return id.isEmpty() ? NONE : BY_ID.get(id);
    }
}

File: datagen/src/main/kotlin/juuxel/adorn/datagen/ConditionType.kt
package juuxel.adorn.datagen

enum class ConditionType(
    private val id: String,
    /** The template for the file path to an external condition file. */
    val separateFilePathTemplate: String? = null,
    /** The path to the template for the text of an external condition file. */
    val separateFileTemplatePath: String? = null,
    val conditionsInFileTemplatePathsByType: Map<String, String> = emptyMap()
) {
    NONE("none"),
    FABRIC("fabric", conditionsInFileTemplatePathsByType = mapOf("load" to "fabric-conditions.json")),
    FORGE("forge", conditionsInFileTemplatePathsByType = mapOf("load" to "forge-conditions.json", "loot-table" to "forge-loot-conditions.json")),
    ;

    companion object {
        private val BY_ID = values().associateBy { it.id }

        fun parse(id: String): ConditionType? =
            if (id.isEmpty()) NONE
            else BY_ID[id]
    }
}
